server/deploy/src/main/scala/com/prisma/deploy/migration/inference/SchemaInferrer.scala

```diff
@@ -161,20 +177,58 @@ case class SchemaInferrerImpl(
           relation.copy(
             name = relationName,
             modelAId = nextModelAId,
-            modelBId = nextModelBId
+            modelBId = nextModelBId,
+            modelAOnDelete = OnDelete.SetNull, //modelAOnDelete,
+            modelBOnDelete = OnDelete.SetNull //modelBOnDelete
           )
         case None =>
           Relation(
             name = relationName,
             modelAId = modelA,
-            modelBId = modelB
+            modelBId = modelB,
+            modelAOnDelete = OnDelete.SetNull, //modelAOnDelete,
+            modelBOnDelete = OnDelete.SetNull //modelBOnDelete
           )
       }
     }
 
     tmp.groupBy(_.name).values.flatMap(_.headOption).toSet
   }
 
+  private def getOppositeField(relationField: FieldDefinition, otherFieldsOnModelBRelatedToModelA: Vector[FieldDefinition]) = {
+    val field = relationField.directive("relation") match {
+      case Some(directive) =>
+        otherFieldsOnModelBRelatedToModelA.find(field =>
+          field.directive("relation") match {
+            case Some(otherDirective) => directive.argument_!("name").value.renderCompact == otherDirective.argument_!("name").value.renderCompact
+            case None                 => false
+        })
+      case None => otherFieldsOnModelBRelatedToModelA.headOption
+    }
+    field
+  }
+
+  private def getOnDeleteFromField(field: FieldDefinition): OnDelete.Value = {
+    field.directive("relation") match {
```

Comment:
you can remove the double nested matches if chain via flatMap:
```
field.directive("relation").flatMap(_.argument("onDelete"") match ...
```
But i think you could just use our existing `directiveArgumentAsString` [method](https://github.com/graphcool/prisma/blob/7b3da2706b9c80a4f17bfa2e571c11618d58f735/server/deploy/src/main/scala/com/prisma/deploy/migration/DataSchemaAstExtensions.scala#L114).

Reply:
Done.